src/common/tusb_fifo.c

```diff
@@ -57,89 +58,365 @@ static void tu_fifo_unlock(tu_fifo_t *f)
 
 bool tu_fifo_config(tu_fifo_t *f, void* buffer, uint16_t depth, uint16_t item_size, bool overwritable)
 {
+  if (depth > 0x8000) return false;               // Maximum depth is 2^15 items
+
   tu_fifo_lock(f);
 
   f->buffer = (uint8_t*) buffer;
   f->depth  = depth;
   f->item_size = item_size;
   f->overwritable = overwritable;
 
-  f->rd_idx = f->wr_idx = f->count = 0;
+  f->max_pointer_idx = 2*depth - 1;               // Limit index space to 2*depth - this allows for a fast "modulo" calculation but limits the maximum depth to 2^16/2 = 2^15 and buffer overflows are detectable only if overflow happens once (important for unsupervised DMA applications)
```

Comment:
using the while() loop for modulo now, I think we don't need to limit the depth to 2^15 any more, right ?